### ISSUE TITLE
Updating asdf-vm not possible when installed with Homebrew

### ISSUE THREAD

**Issue by joschi** [r-darwish/topgrade#367]
**What did you expect to happen?**

It should be possible to update plugins of [asdf-vm](https://asdf-vm.com/) even if [asdf-vm](https://asdf-vm.com/) has been installed with Homebrew.

Refs #334

**What actually happened?**

asdf-vm couldn't be upgraded because the files are not owned by the user running `topgrade`.
Upgrading the plugins would work because the files are stored in `~/.asdf/` but wasn't even tried.

```
# topgrade
[...]
―― 08:55:08 - asdf ―――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――
Update command disabled. Please use the package manager that you used to install asdf to upgrade asdf.

Retry? (y)es/(N)o/(s)hell (Press Ctrl+C again to stop Topgrade)
```

```
# asdf update ; echo $?
Update command disabled. Please use the package manager that you used to install asdf to upgrade asdf.
1
```

```
# asdf plugin-update --all ; echo $?
Updating gradle...
Updating java...
Updating maven...
Updating nodejs...
Updating sbt...
Updating yarn...
0
```

**Additional details:**
- Which operating system or Linux distribution are you using?
  - macOS 10.15.3 (19D76)
  - Darwin localhost 19.3.0 Darwin Kernel Version 19.3.0: Thu Jan  9 20:58:23 PST 2020; root:xnu-6153.81.5~1/RELEASE_X86_64 x86_64
- How did you install Topgrade?
  - Homebrew: `brew install topgrade`

<details><pre>
 2020-03-12T08:14:34.038Z DEBUG topgrade::config > Configuration at /Users/joschi/.config/topgrade.toml
 2020-03-12T08:14:34.038Z DEBUG topgrade::config > Loaded configuration: ConfigFile { pre_commands: None, commands: None, git_repos: None, predefined_git_repos: None, disable: None, remote_topgrades: None, ssh_arguments: None, git_arguments: None, tmux_arguments: None, set_title: None, assume_yes: None, yay_arguments: None, no_retry: None, run_in_tmux: None, cleanup: None, notify_each_step: None, accept_all_windows_updates: None, only: None }
 2020-03-12T08:14:34.038Z DEBUG topgrade         > Version: 4.2.0
 2020-03-12T08:14:34.038Z DEBUG topgrade         > OS: x86_64-apple-darwin
 2020-03-12T08:14:34.038Z DEBUG topgrade         > Args { inner: ["topgrade", "-v"] }
 2020-03-12T08:14:34.038Z DEBUG topgrade         > Binary path: Ok("/usr/local/bin/topgrade")
 2020-03-12T08:14:34.038Z DEBUG topgrade         > Self Update: false
 2020-03-12T08:14:34.038Z DEBUG topgrade::utils  > Detected "/usr/local/bin/git" as "git"
 2020-03-12T08:14:34.038Z DEBUG topgrade::utils  > Detected "/usr/bin/sudo" as "sudo"
 2020-03-12T08:14:34.038Z DEBUG topgrade::utils  > Cannot find "pwsh"
 2020-03-12T08:14:34.039Z DEBUG topgrade::utils  > Cannot find "powershell"
 2020-03-12T08:14:34.039Z DEBUG topgrade::runner > Step "brew"
 2020-03-12T08:14:34.039Z DEBUG topgrade::utils  > Detected "/usr/local/bin/brew" as "brew"

―― 09:14:34 - Brew ―――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――
 2020-03-12T08:14:34.039Z DEBUG topgrade::executor > Running "/usr/local/bin/brew" "update"
Already up-to-date.
 2020-03-12T08:14:35.552Z DEBUG topgrade::executor > Running "/usr/local/bin/brew" "upgrade"
 2020-03-12T08:14:38.890Z TRACE topgrade::executor > Output of "/usr/local/bin/brew" "--repository" "buo/cask-upgrade": Output { status: ExitStatus(ExitStatus(0)), stdout: "/usr/local/Homebrew/Library/Taps/buo/homebrew-cask-upgrade\n", stderr: "" }
 2020-03-12T08:14:38.890Z DEBUG topgrade::executor > Running "/usr/local/bin/brew" "cask" "upgrade"
==> No Casks to upgrade
 2020-03-12T08:14:41.356Z DEBUG topgrade::runner   > Step "MacPorts"
 2020-03-12T08:14:41.356Z DEBUG topgrade::utils    > Cannot find "port"
 2020-03-12T08:14:41.357Z DEBUG topgrade::runner   > Step "nix"
 2020-03-12T08:14:41.357Z DEBUG topgrade::utils    > Cannot find "nix"
 2020-03-12T08:14:41.357Z DEBUG topgrade::runner   > Step "home-manager"
 2020-03-12T08:14:41.357Z DEBUG topgrade::utils    > Cannot find "home-manager"
 2020-03-12T08:14:41.357Z DEBUG topgrade::runner   > Step "asdf"
 2020-03-12T08:14:41.357Z DEBUG topgrade::utils    > Detected "/usr/local/opt/asdf/bin/asdf" as "asdf"

―― 09:14:41 - asdf ―――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――
 2020-03-12T08:14:41.357Z DEBUG topgrade::executor > Running "/usr/local/opt/asdf/bin/asdf" "update"
Update command disabled. Please use the package manager that you used to install asdf to upgrade asdf.
 2020-03-12T08:14:41.379Z DEBUG topgrade::terminal > Desktop notification: asdf failed

Retry? (y)es/(N)o/(s)hell
 2020-03-12T08:14:42.247Z DEBUG topgrade::steps::git > /Users/joschi/.doom.d does not exists
 2020-03-12T08:14:42.247Z DEBUG topgrade::steps::git > Checking if /Users/joschi/.vim is a git repository
 2020-03-12T08:14:42.252Z TRACE topgrade::executor   > Output of "/usr/local/bin/git" "rev-parse" "--show-toplevel": Output { status: ExitStatus(ExitStatus(32768)), stdout: "", stderr: "fatal: Kein Git-Repository (oder irgendeines der Elternverzeichnisse): .git\n" }
 2020-03-12T08:14:42.252Z DEBUG topgrade::steps::git > /Users/joschi/.config/nvim does not exists
 2020-03-12T08:14:42.252Z DEBUG topgrade::steps::git > /Users/joschi/.zshrc is a file. Checking /Users/joschi
 2020-03-12T08:14:42.252Z DEBUG topgrade::steps::git > Checking if /Users/joschi is a git repository
 2020-03-12T08:14:42.257Z TRACE topgrade::executor   > Output of "/usr/local/bin/git" "rev-parse" "--show-toplevel": Output { status: ExitStatus(ExitStatus(32768)), stdout: "", stderr: "fatal: Kein Git-Repository (oder irgendeines der Elternverzeichnisse): .git\n" }
 2020-03-12T08:14:42.257Z DEBUG topgrade::steps::git > /Users/joschi/.tmux does not exists
 2020-03-12T08:14:42.257Z DEBUG topgrade::steps::git > /Users/joschi/.config/fish does not exists
 2020-03-12T08:14:42.257Z DEBUG topgrade::steps::git > /Users/joschi/Library/Preferences/openbox does not exists
 2020-03-12T08:14:42.257Z DEBUG topgrade::steps::git > /Users/joschi/Library/Preferences/bspwm does not exists
 2020-03-12T08:14:42.257Z DEBUG topgrade::steps::git > /Users/joschi/Library/Preferences/i3 does not exists
 2020-03-12T08:14:42.257Z DEBUG topgrade::steps::git > /Users/joschi/Library/Preferences/sway does not exists
 2020-03-12T08:14:42.257Z DEBUG topgrade::runner     > Step "Git repositories"
 2020-03-12T08:14:42.257Z DEBUG topgrade::runner     > Step "zr"
 2020-03-12T08:14:42.257Z DEBUG topgrade::utils      > Detected "/bin/zsh" as "zsh"
 2020-03-12T08:14:42.257Z DEBUG topgrade::utils      > Path "/Users/joschi/.zr" doesn't exist
 2020-03-12T08:14:42.257Z DEBUG topgrade::runner     > Step "antibody"
 2020-03-12T08:14:42.257Z DEBUG topgrade::utils      > Detected "/bin/zsh" as "zsh"
 2020-03-12T08:14:42.257Z DEBUG topgrade::utils      > Cannot find "antibody"
 2020-03-12T08:14:42.257Z DEBUG topgrade::runner     > Step "antigen"
 2020-03-12T08:14:42.257Z DEBUG topgrade::utils      > Detected "/bin/zsh" as "zsh"
 2020-03-12T08:14:42.257Z DEBUG topgrade::utils      > Path "/Users/joschi/.zshrc" exists
 2020-03-12T08:14:42.257Z DEBUG topgrade::utils      > Path "/Users/joschi/antigen.zsh" doesn't exist
 2020-03-12T08:14:42.257Z DEBUG topgrade::runner     > Step "zplug"
 2020-03-12T08:14:42.257Z DEBUG topgrade::utils      > Detected "/bin/zsh" as "zsh"
 2020-03-12T08:14:42.257Z DEBUG topgrade::utils      > Path "/Users/joschi/.zshrc" exists
 2020-03-12T08:14:42.257Z DEBUG topgrade::utils      > Path "/Users/joschi/zplug" doesn't exist
 2020-03-12T08:14:42.257Z DEBUG topgrade::runner     > Step "zinit"
 2020-03-12T08:14:42.257Z DEBUG topgrade::utils      > Detected "/bin/zsh" as "zsh"
 2020-03-12T08:14:42.257Z DEBUG topgrade::utils      > Path "/Users/joschi/.zshrc" exists
 2020-03-12T08:14:42.257Z DEBUG topgrade::utils      > Path "/Users/joschi/.zinit" doesn't exist
 2020-03-12T08:14:42.257Z DEBUG topgrade::runner     > Step "oh-my-zsh"
 2020-03-12T08:14:42.258Z DEBUG topgrade::utils      > Detected "/bin/zsh" as "zsh"
 2020-03-12T08:14:42.258Z DEBUG topgrade::utils      > Path "/Users/joschi/.oh-my-zsh" doesn't exist
 2020-03-12T08:14:42.258Z DEBUG topgrade::runner     > Step "fisher"
 2020-03-12T08:14:42.258Z DEBUG topgrade::utils      > Cannot find "fish"
 2020-03-12T08:14:42.258Z DEBUG topgrade::runner     > Step "tmux"
 2020-03-12T08:14:42.258Z DEBUG topgrade::utils      > Path "/Users/joschi/.tmux/plugins/tpm/bin/update_plugins" doesn't exist
 2020-03-12T08:14:42.258Z DEBUG topgrade::runner     > Step "TLDR"
 2020-03-12T08:14:42.258Z DEBUG topgrade::utils      > Cannot find "tldr"
 2020-03-12T08:14:42.258Z DEBUG topgrade::runner     > Step "rustup"
 2020-03-12T08:14:42.258Z DEBUG topgrade::utils      > Cannot find "rustup"
 2020-03-12T08:14:42.258Z DEBUG topgrade::runner     > Step "cargo"
 2020-03-12T08:14:42.258Z DEBUG topgrade::utils      > Cannot find "cargo-install-update"
 2020-03-12T08:14:42.258Z DEBUG topgrade::runner     > Step "Flutter"
 2020-03-12T08:14:42.258Z DEBUG topgrade::utils      > Cannot find "flutter"
 2020-03-12T08:14:42.258Z DEBUG topgrade::runner     > Step "Go"
 2020-03-12T08:14:42.258Z DEBUG topgrade::utils      > Cannot find "go"
 2020-03-12T08:14:42.258Z DEBUG topgrade::runner     > Step "Emacs"
 2020-03-12T08:14:42.258Z DEBUG topgrade::utils      > Cannot find "emacs"
 2020-03-12T08:14:42.258Z DEBUG topgrade::runner     > Step "opam"
 2020-03-12T08:14:42.258Z DEBUG topgrade::utils      > Cannot find "opam"
 2020-03-12T08:14:42.258Z DEBUG topgrade::runner     > Step "vcpkg"
 2020-03-12T08:14:42.258Z DEBUG topgrade::utils      > Cannot find "vcpkg"
 2020-03-12T08:14:42.258Z DEBUG topgrade::runner     > Step "pipx"
 2020-03-12T08:14:42.258Z DEBUG topgrade::utils      > Cannot find "pipx"
 2020-03-12T08:14:42.258Z DEBUG topgrade::runner     > Step "stack"
 2020-03-12T08:14:42.259Z DEBUG topgrade::utils      > Cannot find "stack"
 2020-03-12T08:14:42.259Z DEBUG topgrade::runner     > Step "tlmgr"
 2020-03-12T08:14:42.259Z DEBUG topgrade::utils      > Cannot find "tlmgr"
 2020-03-12T08:14:42.259Z DEBUG topgrade::runner     > Step "myrepos"
 2020-03-12T08:14:42.259Z DEBUG topgrade::utils      > Cannot find "mr"
 2020-03-12T08:14:42.259Z DEBUG topgrade::runner     > Step "pearl"
 2020-03-12T08:14:42.259Z DEBUG topgrade::utils      > Cannot find "pearl"
 2020-03-12T08:14:42.259Z DEBUG topgrade::runner     > Step "jetpak"
 2020-03-12T08:14:42.259Z DEBUG topgrade::utils      > Cannot find "jetpack"
 2020-03-12T08:14:42.259Z DEBUG topgrade::runner     > Step "vim"
 2020-03-12T08:14:42.259Z DEBUG topgrade::utils      > Detected "/usr/bin/vim" as "vim"
 2020-03-12T08:14:42.261Z TRACE topgrade::executor   > Output of "/usr/bin/vim" "--version": Output { status: ExitStatus(ExitStatus(0)), stdout: "VIM - Vi IMproved 8.1 (2018 May 18, compiled Dec 13 2019 14:45:40)\nIncluded patches: 1-503, 505-680, 682-1312\nCompiled by root@apple.com\nNormal version without GUI.  Features included (+) or not (-):\n+acl               +extra_search      -mouse_netterm     -tag_old_static\n-arabic            -farsi             +mouse_sgr         -tag_any_white\n+autocmd           +file_in_path      -mouse_sysmouse    -tcl\n+autochdir         +find_in_path      -mouse_urxvt       -termguicolors\n-autoservername    +float             +mouse_xterm       +terminal\n-balloon_eval      +folding           +multi_byte        +terminfo\n-balloon_eval_term -footer            +multi_lang        +termresponse\n-browse            +fork()            -mzscheme          +textobjects\n+builtin_terms     -gettext           +netbeans_intg     +textprop\n+byte_offset       -hangul_input      +num64             +timers\n+channel           +iconv             +packages          +title\n+cindent           +insert_expand     +path_extra        -toolbar\n-clientserver      +job               -perl              +user_commands\n-clipboard         +jumplist          +persistent_undo   -vartabs\n+cmdline_compl     -keymap            +postscript        +vertsplit\n+cmdline_hist      +lambda            +printer           +virtualedit\n+cmdline_info      -langmap           -profile           +visual\n+comments          +libcall           +python/dyn        +visualextra\n-conceal           +linebreak         -python3           +viminfo\n+cryptv            +lispindent        +quickfix          +vreplace\n+cscope            +listcmds          +reltime           +wildignore\n+cursorbind        +localmap          -rightleft         +wildmenu\n+cursorshape       -lua               +ruby/dyn          +windows\n+dialog_con        +menu              +scrollbind        +writebackup\n+diff              +mksession         +signs             -X11\n+digraphs          +modify_fname      +smartindent       -xfontset\n-dnd               +mouse             +startuptime       -xim\n-ebcdic            -mouseshape        +statusline        -xpm\n-emacs_tags        -mouse_dec         -sun_workshop      -xsmp\n+eval              -mouse_gpm         +syntax            -xterm_clipboard\n+ex_extra          -mouse_jsbterm     +tag_binary        -xterm_save\n   system vimrc file: \"$VIM/vimrc\"\n     user vimrc file: \"$HOME/.vimrc\"\n 2nd user vimrc file: \"~/.vim/vimrc\"\n      user exrc file: \"$HOME/.exrc\"\n       defaults file: \"$VIMRUNTIME/defaults.vim\"\n  fall-back for $VIM: \"/usr/share/vim\"\nCompilation: gcc -c -I. -Iproto -DHAVE_CONFIG_H   -DMACOS_X_UNIX  -g -O2 -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1      \nLinking: gcc   -L/usr/local/lib -o vim        -lm -lncurses  -liconv -framework Cocoa           \n", stderr: "" }
 2020-03-12T08:14:42.262Z DEBUG topgrade::runner     > Step "Neovim"
 2020-03-12T08:14:42.262Z DEBUG topgrade::utils      > Detected "/usr/local/bin/nvim" as "nvim"
 2020-03-12T08:14:42.262Z DEBUG topgrade::runner     > Step "voom"
 2020-03-12T08:14:42.262Z DEBUG topgrade::utils      > Cannot find "voom"
 2020-03-12T08:14:42.262Z DEBUG topgrade::runner     > Step "NPM"
 2020-03-12T08:14:42.262Z DEBUG topgrade::utils      > Detected "/Users/joschi/.asdf/shims/npm" as "npm"

―― 09:14:42 - Node Package Manager ―――――――――――――――――――――――――――――――――――――――――――――
 2020-03-12T08:14:42.262Z DEBUG topgrade::executor   > Running "/Users/joschi/.asdf/shims/npm" "update" "-g"
 2020-03-12T08:14:42.872Z DEBUG topgrade::runner     > Step "yarn"
 2020-03-12T08:14:42.872Z DEBUG topgrade::utils      > Detected "/Users/joschi/.asdf/shims/yarn" as "yarn"

―― 09:14:42 - Yarn ―――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――
 2020-03-12T08:14:42.872Z DEBUG topgrade::executor   > Running "/Users/joschi/.asdf/shims/yarn" "global" "upgrade" "-s"
 2020-03-12T08:14:43.515Z DEBUG topgrade::runner     > Step "composer"
 2020-03-12T08:14:43.515Z DEBUG topgrade::utils      > Cannot find "composer"
 2020-03-12T08:14:43.515Z DEBUG topgrade::runner     > Step "apm"
 2020-03-12T08:14:43.516Z DEBUG topgrade::utils      > Detected "/usr/local/bin/apm" as "apm"

―― 09:14:43 - Atom Package Manager ―――――――――――――――――――――――――――――――――――――――――――――
 2020-03-12T08:14:43.516Z DEBUG topgrade::executor   > Running "/usr/local/bin/apm" "upgrade" "--confirm=false"
Package Updates Available (0)
└── (empty)
 2020-03-12T08:14:43.961Z DEBUG topgrade::runner     > Step "gem"
 2020-03-12T08:14:43.962Z DEBUG topgrade::utils      > Detected "/usr/bin/gem" as "gem"
 2020-03-12T08:14:43.962Z DEBUG topgrade::utils      > Path "/Users/joschi/.gem" doesn't exist
 2020-03-12T08:14:43.962Z DEBUG topgrade::runner     > Step "Microsoft AutoUpdate"
 2020-03-12T08:14:43.962Z DEBUG topgrade::utils      > Path "/Library/Application Support/Microsoft/MAU2.0/Microsoft AutoUpdate.app/Contents/MacOS/msupdate" doesn't exist
 2020-03-12T08:14:43.962Z DEBUG topgrade::runner     > Step "App Store"
 2020-03-12T08:14:43.962Z DEBUG topgrade::utils      > Detected "/usr/local/bin/mas" as "mas"

―― 09:14:43 - macOS App Store ――――――――――――――――――――――――――――――――――――――――――――――――――
 2020-03-12T08:14:43.962Z DEBUG topgrade::executor   > Running "/usr/local/bin/mas" "upgrade"
Everything is up-to-date
 2020-03-12T08:14:43.997Z DEBUG topgrade::runner     > Step "System upgrade"

―― 09:14:43 - macOS system update ――――――――――――――――――――――――――――――――――――――――――――――
 2020-03-12T08:14:43.998Z DEBUG topgrade::executor   > Running "softwareupdate" "--install" "--all"
Software Update Tool

Finding available software
No updates are available.
 2020-03-12T08:15:15.771Z DEBUG topgrade::runner     > Step "SDKMAN!"
 2020-03-12T08:15:15.771Z DEBUG topgrade::utils      > Detected "/usr/local/bin/bash" as "bash"
 2020-03-12T08:15:15.771Z DEBUG topgrade::utils      > Path "/Users/joschi/.sdkman/bin/sdkman-init.sh" doesn't exist

―― 09:15:15 - Summary ――――――――――――――――――――――――――――――――――――――――――――――――――――――――――
brew: OK
asdf: FAILED
NPM: OK
yarn: OK
apm: OK
App Store: OK
System upgrade: OK
</pre></details>

**Comment by r-darwish** [comment]
Is there any asdf subcommand that we can run in order to find out whether self-update is enabled?

**Comment by joschi** [comment]
@r-darwish I'm not aware of such a (sub-) command.

If it was feasible, we could run `brew list asdf` to find out whether the "asdf" formula was installed. The command would return with a non-zero exit code if it wasn't.

```
# brew list asdf
/usr/local/Cellar/asdf/0.7.7/appveyor.yml
/usr/local/Cellar/asdf/0.7.7/asdf.fish
/usr/local/Cellar/asdf/0.7.7/asdf.sh
/usr/local/Cellar/asdf/0.7.7/ballad-of-asdf.md
/usr/local/Cellar/asdf/0.7.7/bin/asdf
/usr/local/Cellar/asdf/0.7.7/completions/_asdf
/usr/local/Cellar/asdf/0.7.7/CONTRIBUTING.md
/usr/local/Cellar/asdf/0.7.7/defaults
/usr/local/Cellar/asdf/0.7.7/docs/scripts/docsify-edit-on-github.js
/usr/local/Cellar/asdf/0.7.7/docs/ (18 files)
/usr/local/Cellar/asdf/0.7.7/etc/bash_completion.d/asdf.bash
/usr/local/Cellar/asdf/0.7.7/help.txt
/usr/local/Cellar/asdf/0.7.7/lib/commands/ (27 files)
/usr/local/Cellar/asdf/0.7.7/lib/ (3 files)
/usr/local/Cellar/asdf/0.7.7/libexec/private/asdf-exec
/usr/local/Cellar/asdf/0.7.7/lint.sh
/usr/local/Cellar/asdf/0.7.7/release/ (2 files)
/usr/local/Cellar/asdf/0.7.7/SECURITY.md
/usr/local/Cellar/asdf/0.7.7/share/fish/vendor_completions.d/asdf.fish
/usr/local/Cellar/asdf/0.7.7/test/fixtures/ (8 files)
/usr/local/Cellar/asdf/0.7.7/test/ (24 files)
/usr/local/Cellar/asdf/0.7.7/Vagrantfile
/usr/local/Cellar/asdf/0.7.7/VERSION
```

**Comment by r-darwish** [comment]
That's a big Mac specific. Is there any dry run flag to the self-update command?

**Comment by joschi** [comment]
@r-darwish There's no dry-run flag for `asdf update`.

I've created a PR to use a different exit code in case the update isn't possible in contrast to the update failed: asdf-vm/asdf#676

**Comment by r-darwish** [comment]
This is a possible solution, but in this case you'll always see the error message in Topgrade, even if Topgrade won't fail.

If you could make a PR that exposes `asdf_updates_disabled` somehow, perhaps when passing `-h` to the tool, then Topgrade will be able to check if we should run the self upgrade or not and prevent the error message from being displayed.

**Comment by joschi** [comment]
> This is a possible solution, but in this case you'll always see the error message in Topgrade, even if Topgrade won't fail.

I think seeing the message would even be useful if users forgot how they installed asdf-vm in the first place. 😅

> If you could make a PR that exposes `asdf_updates_disabled` somehow

Would it make sense to have a command such as `asdf update --check` without any output on stdout/stderr which exits with exit code 0 if there are updates available and can be applied and exit with exit code >=1 if there are not updates or they cannot be applied.

That way users could run `asdf update --check && asdf update` and `topgrade` could use the exit code of `asdf update --check` to find out whether an update is necessary/possible.

**Comment by joschi** [comment]
Fixed in #374
